woodwork/schema.py

```diff
@@ -327,6 +327,43 @@ def set_time_index(self, new_time_index):
             _check_time_index(self.columns.keys(), new_time_index, self.logical_types.get(new_time_index))
             self._set_time_index_tags(new_time_index)
 
+    def rename(self, columns):
+        """Renames columns in a Schema
+
+        Args:
+            columns (dict[str -> str]): A dictionary mapping columns whose names
+                we'd like to change to the name to which we'd like to change them.
+
+        Returns:
+            woodwork.Schema: Schema with the specified columns renamed.
+
+        Note:
+            Index and time index columns cannot be renamed.
+        """
+        if not isinstance(columns, dict):
+            raise TypeError('columns must be a dictionary')
+
+        for old_name, new_name in columns.items():
+            if old_name not in self.columns:
+                raise KeyError(f"Column to rename must be present. {old_name} cannot be found.")
+            if new_name in self.columns and new_name not in columns.keys():
+                raise ValueError(f"The column {new_name} is already present. Please choose another name to rename {old_name} to or also rename {old_name}.")
+            if old_name == self.index or old_name == self.time_index:
+                raise KeyError(f"Cannot rename index or time index columns such as {old_name}.")
```

Comment:
I don't recall now, but is there a reason we need to prevent renaming index or time index columns?

Reply:
I'm not sure--I can't think of a good reason, though

Reply:
Maybe it had something to do with the complications of how we managed the index and time index before, but this seems like something we could allow now.

Reply:
I'm fine with allowing renaming index columns in the Accessor if you are!

Reply:
I haven't been able to think of a reason why we shouldn't allow it. 

@gsheni Do you have any objections to allowing users to rename the index column or time index column?

Reply:
Nope, I don't have any objections to it, and seems like a useful feature to me.